bridges/primitives/runtime/src/chain.rs

```diff
@@ -236,6 +236,12 @@ where
 pub trait Parachain: Chain {
 	/// Parachain identifier.
 	const PARACHAIN_ID: u32;
+	/// Maximal size of the parachain header.
+	///
+	/// This isn't a strict limit. The relayer may submit larger headers and the
+	/// pallet will accept the call. The limit is only used to compute whether
+	/// the refund can be made.
+	const MAX_HEADER_SIZE: u32;
```

Comment:
Exactly - it may mean multiple things. But you'll probably be computing it in a way I've described in the referenced comment (either theoretical limit w.r.t. current code  OR experimental with some reserve). We may later use it for some other purposes as well. The fact that it is a soft limit for some other code doesn't affect the way it is computed. I can remove this comment if it confuses you and leave it where this "soft" criteria is actually applied, but it won't change the logic or naming or something else.